nio/router/base.py

```diff
@@ -320,6 +324,22 @@ def notify_signals(self, block, signals, output_id):
                             len(signals_to_send),
                             block.name(),
                             receiver_data.block.name()))
+
+                    # if diagnostics are enabled, notify
+                    # through management signal delivery
+                    if self._diagnostics:
+                        self.notify_management_signal(
```

Comment:
This means every time any block notifies a signal we also notify a management signal, right? That is not a scalable solution, we shouldn't double the amount of signals sent through pub/sub in the default implementation. We'll probably want to set up some kind of asynchronous notification type thing to do this instead.